src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java

```diff
@@ -358,6 +359,14 @@ public Void visitEntity(EntityTree tree, Void ignore) {
 
     }
 
+    @Override @DefinedBy(Api.COMPILER_TREE)
+    public Void visitEscape(EscapeTree tree, Void ignore) {
+        hasNonWhitespaceText = true;
```

Comment:
What's the significance of setting this flag here? It doesn't seem to be set in `visitEntity`, for example.

Anyway, when I commented L364 out, no tests failed.

Reply:
It's part of the HTML checking, for [palpable content](https://html.spec.whatwg.org/dev/dom.html#palpable-content)

Some HTML nodes, typically "containers" like `<table>` `<tr>` `<ul>` etc just permit child nodes and whitespace, but no non-white text.  The code here is establishing that the escape characters are treated as non-whitespace characters in any enclosing context.

It is a useful/relevant observation that entities don't set the flag.   I'd have to dig deeper to see if all entities are treated as palpable content in HTML ... the questionable ones being entities like zero-width-space, non-breaking-space etc.  It would be "less wrong" to assume that all entities are not whitespace than to not set the flag at all.